src/odemis/driver/orsay.py

```diff
@@ -2760,3 +2771,242 @@ def terminate(self):
+            if t > tend:
+                raise TimeoutError("Referencing timeout after %g s" % timeout)
+
+        self._updatePosition()
+
+    def _doMoveRel(self, future, shift):
+        new_pos = {"x": self.position.value[0] + shift[0],
+                   "y": self.position.value[1] + shift[1]}
+        self._doMoveAbs(future, new_pos)
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        When the aperture block needs to be moved to a position, or an aperture must be selected.
+        This method is non-blocking.
+        """
+        self._checkMoveAbs(pos)
+        return self._executor.submit(self._doMoveAbs, pos)
+
+    @isasync
+    def moveRel(self, shift):
+        """
+        When the aperture block needs to be moved by a shift, or an aperture must be selected.
+        This method is non-blocking.
+        """
+        self._checkMoveRel(shift)
+        return self._executor.submit(self._doMoveRel, shift)
+
+    @isasync
+    def reference(self, _=None):
+        """
+        reference usually takes axes as an argument. However, the SmarPod references all
+        axes together so this argument is extraneous.
+        """
+        return self._executor.submit(self._doReference)
+
+    def _doReference(self, timeout=10):
+        tend = time.time() + timeout
+        self._hybridAperture.Calibrated.Target = True
+        while self._hybridAperture.Calibrating.Actual == True \
+                or self._hybridAperture.Calibrated != True:
+            time.sleep(0.5)
+            t = time.time()
+            if t > tend:
+                raise TimeoutError("Referencing timeout after %g s" % timeout)
+
+        self._referencedXConnector.update_VA()
+        self._referencedYConnector.update_VA()
+
+    def stop(self, axes=None):
+        """
+        Cancel all queued motions in the executor
+        """
+        t = time.time()
+        self._hybridAperture.IsMovingX.Target = False
+        self._hybridAperture.IsMovingY.Target = False
+
+        logging.debug("Cancelling the executor")
+        self._executor.cancel()
+        # TODO K.K. check for alternative stopping options, speed and actions if this doesn't work
+        # Wait for max 2 seconds for the movement to stop
+        while not self._hybridAperture.IsMovingX.Actual == "False" and not self._hybridAperture.IsMovingY.Actual == "False" \
+                or t + 1 > time.time():
+            time.sleep(0.01)
+        if not self._hybridAperture.IsMovingX.Actual and not self._hybridAperture.IsMovingY.Actual:
+            msg = "The aperture moved is not properly canceled the current status of x is %s and of y is %s" % \
+                  (self._hybridAperture.IsMovingX.Actual, self._hybridAperture.IsMovingY.Actual)
+            logging.debug("Forcing the Actual value of IsMovingX and IsMovingY to false")
+            self._hybridAperture.IsMovingX.Actual = False
+            self._hybridAperture.IsMovingY.Actual = False
+            raise HwError(msg)
```

Comment:
I think a TimeoutError would be more appropriate here. Normally we use HwError to report an error that the hardware expresses (or when we know that the hardware is down/disconnected).